### DISCUSSION TITLE
Android: MarkerView and PointAnnotation show bellow RasterLayer?

### DISCUSSION THREAD

**Discussion by James-Firth** [rnmapbox/maps#1423] [Q&A] [answered]
So I've tried this with both MapboxGL.PointAnnotation and MapboxGL.MarkerView.

- I create an array of markers to display on my map.
- I create an ImageSource and Raster Layer with a remote image URL
- I place those both inside a MapView

Despite the docs saying the MarkerView should _always_ be on top (which it is on iOS), on Android they always seem to appear, then the image source goes over top.

Am I doing something incorrectly or is this a bug?

I'll try and reproduce it with the example apps if that helps, but I'm thinking I need to just revert to using a SymbolLayer on mobile (my web app is using MarkerViews and I was trying to keep the code similar)

Thanks in advance!

**Comment by ferdicus** [comment]
Hi @James-Firth, a reproduction in the example app would definitely help.
Let me know if you need guidance.

Thanks 🙇🏿

**Reply by James-Firth** [reply]
Hi @ferdicus I missed this comment, I'll try and get a reproduction example up and running in the next day or so!

**Reply by ferdicus** [reply]
no hurry, I'm only around Fridays anyways :)

**Reply by James-Firth** [reply]
Oh hm, I tried reproducing it and wasn't able to with the example app and through some testing I've managed to get it working in my app too! (confirming on a physical device shortly)

I believe it was because the content within the `MapboxGL.MarkerView` did not have a `width` property within its `style` AND because I was using the icon from `import { MaterialIcons } from '@expo/vector-icons';` _directly_ as the child, without a view wrapper, which was causing issues (at least in the emulator, hadn't retried on a physical device yet)

**Reply by ferdicus** [reply]
well, then let me mark your reply as the correct answer :)

**Comment by ferdicus** [comment, marked as answer]
> Oh hm, I tried reproducing it and wasn't able to with the example app and through some testing I've managed to get it working in my app too! (confirming on a physical device shortly)
> 
> I believe it was because the content within the `MapboxGL.MarkerView` did not have a `width` property within its `style` AND because I was using the icon from `import { MaterialIcons } from '@expo/vector-icons';` _directly_ as the child, without a view wrapper, which was causing issues (at least in the emulator, hadn't retried on a physical device yet)
